### PR TITLE
Build runtime images and run tests in parallel

### PR DESCRIPTION
**Pull request by chingor13**
By using a Container Builder pipeline, we can reduce our test time from ~37 minutes to ~24 minutes (on CircleCI. The e2e test takes ~12 minutes by itself).

The main cloudbuild.yaml now builds the test runner, php-base image, php-onbuild image, and the gen-dockerfile builder.

Each of these images' tests are also run as soon as their respective image is built.

### CODE REVIEW

**Comment by chingor13** [comment]
Parallelizing the build uncovered an race condition using the structure_test image in parallel. https://github.com/GoogleCloudPlatform/runtimes-common/pull/169

File: cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    id: php71-extensions-build
+  - name: gcr.io/cloud-builders/docker
+    args: ['run', 'gcr.io/$PROJECT_ID/php71-extensions:$_TAG', 'vendor/bin/phpunit']
+    dir: testapps/php71_extensions
+    waitFor: ['php71-extensions-build']
+
+images:
+  - gcr.io/$PROJECT_ID/php-base:$_TAG
+  - gcr.io/$PROJECT_ID/php:$_TAG
+  - gcr.io/$PROJECT_ID/php-base:$_TAG
```
Comment (tmatsuo) [inline cloudbuild.yaml]: gen-dockerfile?

Comment (chingor13) [inline cloudbuild.yaml]: will fix

File: builder/gen-dockerfile/src/create_dockerfile.php [hunk 2]
```diff
@@ -18,8 +18,10 @@
 require_once __DIR__ . '/../vendor/autoload.php';
 require_once __DIR__ . '/GenFiles.php';
 
-$genFiles = new GenFiles();
-if ($argc === 2) {
+$genFiles = $argc >= 3
```
Comment (chingor13) [inline builder/gen-dockerfile/src/create_dockerfile.php]: We can use option parsing in a separate PR

Comment (tmatsuo) [inline builder/gen-dockerfile/src/create_dockerfile.php]: I have a change with Symfony\Console

**Comment by chingor13** [comment]
We can also run the end to end test in parallel with other tests if it always runs. It's currently possible to skip the e2e tests, but both circle and our release builder are configured to run them.

**Comment by chingor13** [comment]
@tmatsuo can you take a look at the last few changes since the merge from master. Switched to using the Symfony Console CommandTester.

**Comment by tmatsuo** [comment]
LGTM

**Comment by tmatsuo** [comment]
Oh wait. Do we need to change `builder/php-latest.yaml`?

**Comment by chingor13** [comment]
I left the base image as the first argument and made the workspace an option.

**Pull request merged**